src/main/java/crawlercommons/filters/basic/BasicURLNormalizer.java

```diff
@@ -305,7 +519,7 @@ private String unescapePath(String path) {
         // Traverse over all encoded groups
         while (matcher.find()) {
             // Append everything up to this group
-            sb.append(path.substring(end + 1, matcher.start()));
+            sb.append(path, end + 1, matcher.start());
```

Comment:
In case anyone is wondering about this change, these lines are equivalent. Intellij automatically suggests this refactoring, which avoids one extra function call.